nautobot/core/tests/test_imports.py

```diff
@@ -0,0 +1,415 @@
+            defined.add(name)
+
+    return defined
+
+
+def find_already_imported_in_apps(apps_dir):
+    """Find all functions and classes already imported in the nautobot.apps directory."""
+    imported = set()
+    for fname in os.listdir(apps_dir):
+        if not fname.endswith(".py"):
+            continue
+        fpath = os.path.join(apps_dir, fname)
+        if not os.path.isfile(fpath):
+            continue
+        with open(fpath, "r") as f:
+            try:
+                tree = ast.parse(f.read(), filename=fpath)
+            except (SyntaxError, UnicodeDecodeError, ValueError):
+                continue
+        for node in ast.walk(tree):
+            if isinstance(node, ast.ImportFrom):
+                for alias in node.names:
+                    imported.add(alias.asname or alias.name)
+            elif isinstance(node, ast.Import):
+                for alias in node.names:
+                    imported.add(alias.asname or alias.name)
+    return imported
+
+
+def find_all_python_files(root_dir):
+    """Recursively find all Python files under a given root directory, excluding blacklisted files/folders."""
+    py_files = []
+    for root, _, files in os.walk(root_dir):
+        # Skip blacklisted folders
+        if any(BLACKLIST_FILE_REGEX.match(part) for part in os.path.normpath(root).split(os.sep)):
+            continue
+        for fname in files:
+            if not fname.endswith(".py"):
+                continue
+            # Skip blacklisted files
+            if BLACKLIST_FILE_REGEX.match(fname):
+                continue
+            py_files.append(os.path.join(root, fname))
+    return py_files
+
+
+def main():
+    """Main function to find missing imports in nautobot.apps."""
+    nautobot_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+    remove_dir_prefix = os.path.dirname(nautobot_dir) + "/"
+
+    apps_dir = os.path.join(nautobot_dir, "apps")
+    imported_symbols = find_already_imported_in_apps(apps_dir)
+    py_files = find_all_python_files(nautobot_dir)
+    output = []
+
+    for source_file in py_files:
+        # Skip files in nautobot/apps
+        if os.path.commonpath([source_file, apps_dir]) == apps_dir:
+            continue
+        if is_blacklisted_file(source_file):
+            continue
+        defined_symbols = find_func_or_class(source_file, remove_dir_prefix)
+        missing_exports = defined_symbols - imported_symbols
+        if missing_exports:
+            source_file = source_file.replace(remove_dir_prefix, "").replace(".py", "").replace("/", ".")
+            output.append(f"\nfrom {source_file} import (")
+            for symbol in sorted(missing_exports):
+                output.append(f"  {symbol},")
+            output.append(")")
+    return "\n".join(output)
+
+
+class ImportTestCase(TestCase):
+    """Test case to ensure nautobot.apps contains all relevant imports."""
+
+    def test_imports(self):
+        output = main()
```

Comment:
Out of curiosity, how long does this test take to run?

Reply:
```
Ran 1 test in 0.630s
```